python/revng/restapi/api.py

```diff
@@ -0,0 +1,84 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import json
+from typing import TYPE_CHECKING
+
+from flask import Blueprint, jsonify, request, current_app
+
+from ariadne import graphql
+from ariadne.constants import PLAYGROUND_HTML
+from ariadne.file_uploads import combine_multipart_data
+from ariadne.contrib.tracing.apollotracing import ApolloTracingExtension
+
+from .producer_data_loader import ProducerDataLoader
+from .schema import SchemafulManager
+
+
+if TYPE_CHECKING:
+    from flask.ctx import _AppCtxGlobals
+
+    class FlaskGlobals(_AppCtxGlobals):
+        manager: SchemafulManager
+
+    g = FlaskGlobals()
+else:
+    from flask import g
+
+
+api_blueprint = Blueprint("api", __name__)
+
+
+def json_response(data=None):
+    if data is None:
+        data = {}
+    return jsonify(data)
+
+
+def json_error(message, http_code=404):
+    data = {
+        "error": message,
+    }
+    return jsonify(data), http_code
+
+
+PLAYGROUND_CONFIG = """
+    settings: {
+        ["request.credentials"]: 'same-origin'
+    }
+"""
+NEW_PLAYGROUND_HTML = PLAYGROUND_HTML.replace(
+    "// options as 'endpoint' belong here", PLAYGROUND_CONFIG
+)
+
+
+@api_blueprint.route("/graphql", methods=["GET"])
+def graphql_playground():
+    return NEW_PLAYGROUND_HTML, 200
+
+
+@api_blueprint.route("/graphql", methods=["POST"])
+async def graphql_server():
+    if request.content_type.startswith("multipart/form-data;"):
+        operations = request.form.get("operations")
+        req_map = request.form.get("map")
+        if operations is not None and req_map is not None:
+            data = combine_multipart_data(
+                json.loads(operations), json.loads(req_map), request.files  # type: ignore
```

Comment:
What are these `# type: ignore` comments?

Reply:
* If we commit directives for linters and the like, they need to be checkable by everyone, if it's something that just your editor tells you, we're sad.
  Is this why you want `flake8`?
  Do you want to add `mypy` to `revng check-conventions`?
* What do you think of assigning it to an explicitly annotated temporarily variable?

Reply:
1. Right now I just have it locally, but while we are at it might as well add `mypy`.
2. Might not be a bad idea, should improve readability